File: im-chat-master/chat-room-vue/src/lim/pack/GroupMessagePack.ts
import Beans from '../common/utils';

export class GroupMessagePack {
    appId: number;
    messageId: string;
    fromId?: string;
    groupId?: string;
    messageRandom?: number;
    messageTime?: Date;
    messageBody?: string;
    FileContent?:object;
    messageKey?: string;
    type: number;
    mediaContent?:Object;


    constructor(appId: number) {
        this.messageId = Beans.uuid();
        this.appId = appId;
        this.messageRandom = this.RangeInteger(0, 10000);
        this.messageTime = new Date();
    }

    RangeInteger(min: number, max: number) {
        const range = max - min
        const value = Math.floor(Math.random() * range) + min
        return value
    }


    buildTextMessagePack(fromId: string, groupId: string, text: string,type:number) {
        this.fromId = fromId;
        this.groupId = groupId;
        this.messageBody = text;
        this.type = type
    }

    buildFileMessagePack(fromId: string, groupId: string, text: object,type:number) {
        this.fromId = fromId;
        this.groupId = groupId;
        this.FileContent = text;
        this.type = type
    }

    buildMediaMessagePack(fromId: string, groupId: string, type:number,key:string,mediaContent:Object) {
        this.fromId = fromId;
        this.groupId = groupId;
        this.type = type;
        this.messageKey = key;
        this.mediaContent = mediaContent
    }

    buildCustomerMessagePack(fromId: string, groupId: string, type: number, obj: any) {
        this.fromId = fromId;
        this.groupId = groupId;
        let body = { type: type, content: obj }
        this.messageBody = Beans.json(body);
    }
}

File: im/im-system/im-codec/src/main/java/com/lld/im/codec/pack/group/GroupMessagePack.java
package com.lld.im.codec.pack.group;

import lombok.Data;

import java.util.Date;

/**
 * @author: Chackylee
 * @description: 群聊消息分发报文
 **/
@Data
public class GroupMessagePack {

    //客户端传的messageId
    private String messageId;

    private String messageKey;

    private String fromId;

    private String groupId;

    private int messageRandom;

    private Date messageTime;

    private long messageSequence;

    private String messageBody;
    /**
     * 这个字段缺省或者为 0 表示需要计数，为 1 表示本条消息不需要计数，即右上角图标数字不增加
     */
    private int badgeMode;

    private Long messageLifeTime;

    private Integer appId;

}
